xds/src/main/java/io/grpc/xds/XdsClientMetricReporterImpl.java

```diff
@@ -143,7 +143,13 @@ void reportCallbackMetrics(BatchRecorder recorder, XdsClient xdsClient) {
       Map<XdsResourceType<?>, Map<String, ResourceMetadata>> metadataByType =
           getResourceMetadataCompleted.get(10, TimeUnit.SECONDS);
 
-      computeAndReportResourceCounts(metadataByType, callback);
+      ListenableFuture<Map<XdsResourceType<?>, Map<String, String>>>
+              getResourceAuthorityCompleted = xdsClient.getSubscribedResourcesAuthoritySnapshot();
+
+      Map<XdsResourceType<?>, Map<String, String>> authorityByType =
+              getResourceAuthorityCompleted.get(10, TimeUnit.SECONDS);
```

Comment:
When using futures, you'd want to start the operations and only then wait for the completions. That way you only pay the latency cost once.